tests/main_test.go

```diff
@@ -239,11 +239,27 @@ func (c *TestContext) WithAliceBobAndCharlieSyncing(t *testing.T, callback func(
 }
 
 // CreateNewEncryptedRoom calls creator.MustCreateRoom with the correct m.room.encryption state event.
-func (c *TestContext) CreateNewEncryptedRoom(t *testing.T, creator *client.CSAPI, preset string, invite []string) (roomID string) {
+
+// m_room_encryptionContent specifies the content of the `m.room.encryption`
+// event. It may be nil, in which case the standard algorithm
+// ("m.megolm.v1.aes-sha2") is used, and the rotation periods are omitted,
+// meaning the defaults will be used.
+func (c *TestContext) CreateNewEncryptedRoom(
+	t *testing.T,
+	creator *client.CSAPI,
+	preset string,
+	invite []string,
+	m_room_encryptionContent map[string]interface{},
```

Comment:
No underscores please, camelCase.

Also, at this point it would be good to get this as a functional option rather than just another param. See https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis - I suggest a function signature of `func(reqBody map[string]any)` for the option. Both Complement and Complement-Crypto make heavy use of this pattern, so implementing it yourself here would be a good way to learn why :D